### ISSUE TITLE
SNOW-779392: Unpacking a string is disallowed

### ISSUE THREAD

**Issue by tekumara** [snowflakedb/snowflake-connector-python#1502]
```python
import snowflake.connector

def load(conn: snowflake.connector.SnowflakeConnection):
    with conn.cursor() as cur:
        res = cur.execute(f"""SELECT 'Hello World'""")
        if res:
            reveal_type(res)
            for (cell,) in res:
                print(cell)
```

mypy fails with:
```
test.py:7: note: Revealed type is "Union[snowflake.connector.cursor.SnowflakeCursor, builtins.dict[builtins.str, Any]]"
test.py:8: error: Unpacking a string is disallowed  [misc]
```

At runtime the actual type is `snowflake.connector.cursor.SnowflakeCursor`.

Failing since v3.0.1. In v3.0.0 the revealed type is `Any` and so there is no failure.

**Comment by sfc-gh-aling** [comment]
hey @tekumara , thanks again for your feedback. This is an open source project so we would love to see if you could help contribute to it! Do you creating a PR for this, we can help review the PR.

**Comment by sfc-gh-achandrasekaran** [comment]
@tekumara let us know if you want to put out a PR here. Otherwise we will look into it in a few weeks

**Comment by tekumara** [comment]
The good news is, I've tested this with the fix from #1501 and it's now resolved! So I'll close this.
